packages/stable/src/api/geospatial/types.ts

```diff
@@ -0,0 +1,245 @@
+import { InternalId, Limit, ExternalId } from '@cognite/sdk-core';
+
+import { Geometry, GeoJSON } from 'geojson';
+
+export { GeoJSON };
+
+export type GeometryType = 'WKT' | 'GEOJSON';
+
+type GeometryPropertyType =
+  | 'GEOMETRY'
+  | 'POINT'
+  | 'LINESTRING'
+  | 'POLYGON'
+  | 'MULTIPOINT'
+  | 'MULTILINESTRING'
+  | 'MULTIPOLYGON'
+  | 'GEOMETRYCOLLECTION'
+  | 'GEOMETRYZ'
+  | 'POINTZ'
+  | 'LINESTRINGZ'
+  | 'POLYGONZ'
+  | 'MULTIPOINTZ'
+  | 'MULTILINESTRINGZ'
+  | 'MULTIPOLYGONZ'
+  | 'GEOMETRYM'
+  | 'POINTM'
+  | 'LINESTRINGM'
+  | 'POLYGONM'
+  | 'MULTIPOINTM'
+  | 'MULTILINESTRINGM'
+  | 'MULTIPOLYGONM'
+  | 'GEOMETRYCOLLECTIONM'
+  | 'GEOMETRYZM'
+  | 'POINTZM'
+  | 'LINESTRINGZM'
+  | 'POLYGONZM'
+  | 'MULTIPOINTZM'
+  | 'MULTILINESTRINGZM'
+  | 'MULTIPOLYGONZM'
+  | 'GEOMETRYCOLLECTIONZM';
+
+export type GeospatialPropertyType =
+  | 'STRING'
+  | 'LONG'
+  | 'DOUBLE'
+  | 'BOOLEAN'
+  | 'TIMESTAMP'
+  | GeometryPropertyType;
+
+export type GeospatialFeatureTypeProperty =
+  | {
+      type: Exclude<GeospatialPropertyType, GeometryPropertyType | 'STRING'>;
+      description?: string;
+      optional?: boolean;
+    }
+  | {
+      type: GeometryPropertyType;
+      srid?: number;
+      description?: string;
+      optional?: boolean;
+    }
+  | {
+      type: 'STRING';
+      size: number;
+      description?: string;
+      optional?: boolean;
+    };
+
+type Properties = Record<string, GeospatialFeatureTypeProperty>;
+
+export interface Geospatial {
+  featureTypeAPI: unknown;
+  featureAPI: unknown;
+}
+
+export interface GeospatialFeature extends ExternalId {
+  [property: string]:
+    | string
+    | number
+    | boolean
+    | Date
+    | { wkt: string }
+    | Geometry;
+}
+
+export interface GeospatialFeatureResponse extends GeospatialFeature {
+  createdTime: Date;
+  lastUpdatedTime: Date;
+}
+
+export interface FeatureType extends ExternalId, InternalId {
+  properties: Properties & {
+    _created_at: { type: 'LONG' };
+    _updated_at: { type: 'LONG' };
+    _external_id: { type: 'STRING'; size: 32 };
+  };
+  searchSpec?: {
+    [indexName: string]: {
+      properties:
+        | string[]
+        | ['_created_at']
```

Comment:
API returns `externalId`, `lastUpdatedTime` and `createdTime`

Reply:
@pawelzwolenskicognite This is featureType and in the example this is what is returned after creation.
Can you please confirm if this is correct or not ?
<img width="481" alt="Screenshot 2022-01-04 at 12 44 35 PM" src="https://user-images.githubusercontent.com/86822653/148054445-3dc0e508-3524-4c2b-9a89-646c0abc87d3.png">

Reply:
looks like the example is outdated - we will fix it, good catch!

btw. that should be tested in some automatic tests?

Reply:
@saagar-cognite the example was fixed (refresh https://github.com/cognitedata/service-contracts/pull/1323)